conda/plan.py

```diff
@@ -290,6 +290,9 @@ def ensure_linked_actions(dists, prefix, index=None, force=False, always_copy=Fa
         if not extracted_in:
             actions[inst.EXTRACT].append(dist)
 
+        # find_new_location returns either a `pkgs` directory or the filename.
+        if not fetched_in.endswith('.tar.bz2'):
+            fetched_in = join(fetched_in, install._dist2filename(dist))
```

Comment:
I'm not convinced this is correct. `fetched_in` is supposed to return a `.tar.bz2` filename or None if the file doesn't exist. if it's not, there's a bug somewhere else, and we need to find it.